contracts/contracts/harvest/Harvester.sol

```diff
@@ -0,0 +1,337 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity ^0.8.0;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { SafeMath } from "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+
+import { StableMath } from "../utils/StableMath.sol";
+import { Initializable } from "../utils/Initializable.sol";
+import { Governable } from "../governance/Governable.sol";
+import { IVault } from "../interfaces/IVault.sol";
+import { IOracle } from "../interfaces/IOracle.sol";
+import { IStrategy } from "../interfaces/IStrategy.sol";
+import { IUniswapV2Router } from "../interfaces/uniswap/IUniswapV2Router02.sol";
+import "../utils/Helpers.sol";
+
+contract Harvester is Initializable, Governable {
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+    using StableMath for uint256;
+
+    // Tokens that should be swapped for stablecoins
+    address[] public swapTokens;
+
+    // Address of Uniswap
+    address public uniswapAddr = address(0);
```

Comment:
We should have this be per token, since we can get better deals on different routers.

We should also have the fee be per token, but in this contract, not on the strategies. That's not really something the strategies need to be concerned about.

Both should be in a struct mapped to the token. Also in the same struct should be the max slippage allowed for selling that token.

Reply:
Good deal, I will add reward liquidation limit - a per token setting of how much token can be sold at once to this struct as well